src/Tasks.UnitTests/Copy_Tests.cs

```diff
@@ -2052,11 +2051,32 @@ public class CopyHardAndSymbolicLink_Tests
         [Fact]
         public void CopyWithHardAndSymbolicLinks()
         {
+            // Workaround: For some reason when this test runs with all other tests we are getting
+            // the incorrect result from CreateHardLink error message (a message associated with
+            // another test). Calling GetHRForLastWin32Error / GetExceptionForHR seems to clear
+            // out the previous message and allow us to get the right message in the Copy task.
+            int errorCode = Marshal.GetHRForLastWin32Error();
+            Marshal.GetExceptionForHR(errorCode);
+            
             string sourceFile = FileUtilities.GetTemporaryFile();
-            string temp = Path.GetTempPath();
+            const string temp = @"d:\\temp";
```

Comment:
How to define a different destination folder from source folder applicable for all the OS?

Reply:
This test was sorta doing that previously. FileUtilities.GetTemporaryFile() should get a random file in the temp directory, and this initializes destFolder to be a different folder (a subfolder). If you want them fully separate, you could put the source file in a subfolder under temp and the destination in a separate subfolder under temp.

Reply:
Sorry, I didn't descipe this issue well before. The UseHardlinksIfPossible doesn't work when the source and target are on different drives. This test is for failing to create hard link in different driveS, and fallback to create a symbolic link. I'm wondering how to create different source and destination folders with different drivers applicable for all the OS.

Reply:
Ah, clever. I'm not aware of a similar system on mac/linux with multiple drives. I was tempted to suggest writing into protected memory, but that'll succeed if you have admin privileges, which is bad. I can't think of any other good options off the top of my head. @rainersigwald?